### PR TITLE
[rclpy_examples] QoS examples

### PR DESCRIPTION
**Pull request by mikaelarguedas**
Add reliable/best effort talker listener example

### CODE REVIEW

File: rclpy_examples/listener_qos_py.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+def chatter_callback(msg):
+    print('I heard: [%s]' % msg.data)
+
+
+def main(argv=sys.argv[1:]):
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-r', '--reliability', type=int, default=0, choices=[0, 1],
+        help='0: reliable, 1: best effort')
```
Comment (dirk-thomas) [inline rclpy_examples/listener_qos_py.py]: When being called `reliability` I would expect the value `1` to actually be reliable.

Comment (mikaelarguedas) [inline rclpy_examples/listener_qos_py.py]: thanks for pointing it out, fixed in https://github.com/ros2/examples/pull/132/commits/1ce623944a512c9c2f86a1d0ba6583856f4a2619

File: rclpy_examples/listener_qos_py.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+def chatter_callback(msg):
+    print('I heard: [%s]' % msg.data)
+
+
+def main(argv=sys.argv[1:]):
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-r', '--reliability', type=int, default=0, choices=[0, 1],
+        help='1: reliable, 0: best effort')
```
Comment (dirk-thomas) [inline rclpy_examples/listener_qos_py.py]: Are there any other options planned? If not I would suggest using a boolean flag `--reliable` instead.

**Comment by dirk-thomas** [comment]
Does it make sense to create new scripts rather then extending the already existing ones?

**Comment by mikaelarguedas** [comment]
I see your point, this comes down to how simple do we want the examples to be, should the basic pub/sub examples expose how to configure the Quality of Service? or does it make sense to have another script showing how to handle QoS on top of the basic example?
IMO the code in this PR could be extended to expose more quality of service settings in the future and could be used as a full QoS example/tutorial and should stay dissociated from the minimalist pub/sub.
(this PR also has the goal of giving a starting point for https://github.com/ros2/demos/issues/85).

**Comment by mikaelarguedas** [comment]
thanks

**Comment by mikaelarguedas** [comment]
If nobody has strong opinions about https://github.com/ros2/examples/pull/132#issuecomment-257907246 I'm going to merge this as is

**Comment by dirk-thomas** [comment]
Sure, go ahead with the merge.

**Pull request merged**